File: c_plus_plus/1029.cpp
#include <iostream>

int NUM_CALLS;

int fibo(int n) {
    NUM_CALLS++;
    if (n == 0) return 0;
    if (n == 1) return 1;
    else return fibo(n-1) + fibo(n-2);
}

int main() {
    int n, x;
    std::cin >> n;
    for (int i = 0; i < n; i++) {
        std::cin >> x;
        NUM_CALLS = 0;
        int resultado = fibo(x);
        std::cout << "fib(" << x << ") = " << NUM_CALLS - 1 << " calls = " << resultado << std::endl;
    }
    return 0;
}

File: 1029/1029.rs
use std::io;

fn fibonacci(n : u32, num_calls : &mut u32) -> u32{
    *num_calls += 1;

    match n {
        0 => 0,
        1 => 1,
        _ => fibonacci(n-1, num_calls) + fibonacci(n-2, num_calls),
    }
}

fn main(){
    let mut entrada : String = String::new();
    io::stdin().read_line(&mut entrada).expect("msg");

    let t : usize = entrada.trim().parse().expect("msg");

    for _ in 0..t{
        let mut entrada : String = String::new();
        io::stdin().read_line(&mut entrada).expect("msg");

        let x : u32 = entrada.trim().parse().expect("msg");

        let mut num_calls: u32 = 0;
        let res : u32 = fibonacci(x, &mut num_calls);

        println!("fib({}) = {} calls = {}", x, num_calls - 1, res);
    }
}